### DISCUSSION TITLE
Automation of 2015 Coffeetool R5

### DISCUSSION THREAD

**Discussion by stef042** [artisan-roaster-scope/artisan#1118] [Q&A] [answered]
Hi all
Maybe I'm wrong here but here's my issue:
I am completely new to coffee roasters and a friend asked me for some tech support.

I want to read and send data to a [2015 Coffeetool R5 model](https://coffeetool.gr/product/r5/). The machine has a USB connector only and it seems that it's possible to export data to an XLS.
However, I would like to use Artisan to log data in realtime and -if possible- to control the roaster with the equipment that's already built-in.

It seems that it isn't like plug & play and I read about several possibilities to achieve what I want. I think that the Arduino UNO & TC4C solution or MODBUS is something I should go for.

Any experiences with that or similar machine type, proposals or other ideas?
Any help would be appreciated!

Thanks,
Stefan

**Comment by poundy** [comment, marked as answer]
There are two approaches here.

1) you find out what the USB connector offers; if it offers a connection to the PIDs (that do the temperature readout) that are already in the machine, you figure out what registers you need to read, and use Artisan to read those.

2) you scrap all that and implement a second set of probes to measure and report temperature with Artisan.  Here you have multiple choices as well, you can implement your own (Arduino/TC4 and thermocouple measurement ICs) or you can use something like Phidgets, a simple, modular solution that is flexible and simple to implement.

 it's really your choice which path you go down. Often, if the machine is well documented, the first option is simplest and most practical.  Given that Artisan already has some support for Coffeetool machines https://artisan-scope.org/machines/coffeetool/ there's a chance the PLC registers are already similar to what the current models use, and that would be my suggestion, chase as much doco from the manufacturer that they will give you for an older machine and then use modbus over usb to get the data.

**Comment by stef042** [comment]
Thank you for your helpful reply, poundy!

After I was able to take a look inside machine and control panel I found some helpful parts and connectors. Newer models have a RJ45 port accessible below the USB port on the control panel box, but inside I could see that the connection between control panel and PLC is already network based.


![PXL_20230303_192656535](https://user-images.githubusercontent.com/119729267/222959509-cb25a68d-091d-418a-b12b-31f2eac3e0b3.jpg)
![PXL_20230303_193235747](https://user-images.githubusercontent.com/119729267/222959535-c521d6b5-c4c5-423f-8896-0a0b40934e9c.jpg)

The Bejier X2 connects to the Allen-Bradley PLC via Ethernet. Coffetool R series is supported by Artisan. I need to figure out the commands to switch on/off cooling fan and tray and add the buttons to Artisan. So far so good. I didn't get to testing yet, but I am 95% sure that a direct connection to the PLC with Artisan will be successful. What I don't know yet is, if it is possible to have the original control panel AND a computer connected to the PLC via a network switch, because I'd like to test the setup with Artisan first and keep the machine working with the original panel.
Any experiences with having two control devices connected to an Allen Bradley 820 via network? Is that a stupid idea?

**Reply by stef042** [reply]
UPDATE: Added a switch to connect original Beijer X2 control display, Laptop running Artisan to PLC. Everything worked well. Next steps will be adding preheat, switching on/off cooling tray and fan.
I don't know yet where preheat process including ignition is configured, any ideas?

**Reply by poundy** [reply]
the fact they use a LAN between the PLC network and the Bejier X2 control box should simply mean you can do as you say, connect the PLC, the control box, and a PC to a switch and it should all just work. I have not seen two "controllers" talking to the same PLC network, but I would expect you'd fairly quickly see if there were issues (eg if one device talked to the PLC in such a way that blocked the other one) so even in your testing phase you'd likely see it work, or it break... I would think that you're unlikely to see it break in some weird condition that happened weeks after thinking this was working fine.... But with all things "undocumented", I am guessing and your experience may vary, and the risk is all yours to accept. Keep us updated on your progress though !

**Reply by MAKOMO** [reply]
You might make use of the Artisan build-in MODBUS scanner (Scanner button in the MODBUS tab) to look for the register numbers with the data of interest. Note however, that in the current Artisan v2.8.2 the scanner is broken. The current Continuous build contains already the fix.
